control-plane-operator/controllers/hostedcontrolplane/mcs/params.go

```diff
@@ -0,0 +1,48 @@
+package mcs
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/config"
+	"github.com/openshift/hypershift/support/globalconfig"
+)
+
+type MCSParams struct {
+	OwnerRef       config.OwnerRef
+	RootCA         *corev1.Secret
+	CombinedCA     *corev1.ConfigMap
+	PullSecret     *corev1.Secret
+	DNS            *configv1.DNS
+	Infrastructure *configv1.Infrastructure
+	Network        *configv1.Network
+	Proxy          *configv1.Proxy
+	InstallConfig  *globalconfig.InstallConfig
+}
+
+func NewMCSParams(hcp *hyperv1.HostedControlPlane, rootCA, pullSecret *corev1.Secret, combinedCA *corev1.ConfigMap, apiServerAddress string, apiServerPort int32) *MCSParams {
+	dns := globalconfig.DNSConfig()
+	globalconfig.ReconcileDNSConfig(dns, hcp)
```

Comment:
Missing error checks (why does the signature include an err return if its always nil?)

Reply:
it shouldn't, I'll fix it